salt/modules/test.py

```diff
@@ -628,3 +630,44 @@ def false_():
         salt '*' test.false
     '''
     return False
+
+
+def raise_exception(name, *args, **kwargs):
+    '''
+    Raise an exception. Built-in exceptions and those in ``salt.exceptions``
+    can be raised by this test function. If no matching exception is found,
+    then no exception will be raised and this function will return ``False``.
+
+    This function is designed to test Salt's exception and return code
+    handling.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' test.raise_exception TypeError "An integer is required"
+        salt '*' test.raise_exception salt.exceptions.CommandExecutionError "Something went wrong"
+    '''
+    def _is_exc(cls):
+        for base in cls.__bases__:
+            if base is BaseException:
+                break
+            else:
+                return _is_exc(base)
+        else:
+            return False
+        return True
+
+    try:
+        if name.startswith('salt.exceptions.'):
+            exc = getattr(salt.exceptions, name[16:])
+        else:
+            exc = getattr(builtins, name)
+        if _is_exc(exc):
+            raise exc(*args, **salt.utils.args.clean_kwargs(**kwargs))
+        else:
+            log.error('%s is not an exception', name)
```

Comment:
This is not the most intuitive error message, IMHO. As a user, I wouldn't know what to do if I saw this.

Reply:
It would be accompanied by a line number in the log file. This is a function designed to be used for testing, so I didn't think we needed something that was all that verbose. The idea is that if you are running this function, you're doing it for a reason. I don't think it's something that 99.99% of non-developer users are ever going to see.

What did you have in mind?

Reply:
Ah, I failed to recognize that this was in the test module. Yes, your counter-point is sound.